balance_check/providers/gamestop.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import requests
+from bs4 import BeautifulSoup
+from balance_check import logger, captcha_solver, config
+from balance_check.providers import BalanceCheckProvider
+from balance_check.validators.gift_card import Merchant, GiftCardSchema
+
+
+class GameStop(BalanceCheckProvider):
+    def __init__(self):
+        super().__init__()
+
+        self.website_url = "https://www.gamestop.com/profiles/valuelookup.aspx"
+        self.schema = GiftCardSchema([Merchant.GameStop])
+
+    def scrape(self, fields):
+        session = requests.Session()
+        session.headers.update({
+            "User-Agent": config.USER_AGENT
+        })
+
+        fields["X-Requested-With"] = "XMLHttpRequest"
+
+        logger.info("Fetching balance check page")
+
+        resp = session.get(self.website_url)
+        if resp.status_code != 200:
+            logger.critical("Failed to GET website (status code {})".format(resp.status_code))
```

Comment:
Raise an exception instead of logging and bailing (see newer version of the example).
```suggestion
            raise RuntimeError("Failed to GET website (status code {})".format(resp.status_code))
```